diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2021-01-13 17:41:39.000000000 +0000
+++ b/CMakeLists.txt	2021-04-28 21:38:32.140929366 +0000
@@ -209,6 +209,11 @@
     ${quaternion_SRCS} ${quaternion_QRC_SRC} ${quaternion_QM}
     ${quaternion_WINRC} ${${PROJECT_NAME}_MAC_ICON})
 
+if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.16.0"
+        AND NOT CMAKE_CXX_COMPILER_ID STREQUAL GNU) # https://bugzilla.redhat.com/show_bug.cgi?id=1721553
+    target_precompile_headers(${PROJECT_NAME} PRIVATE <room.h>)
+endif ()
+
 target_link_libraries(${PROJECT_NAME}
     QMatrixClient Qt5::Widgets Qt5::Quick Qt5::Qml Qt5::Gui Qt5::Network)
 target_compile_definitions(${PROJECT_NAME} PRIVATE
